src/systems/systems/src/pneumatic/valve.rs

```diff
@@ -0,0 +1,690 @@
+use std::f64::consts::PI;
+
+use crate::{
+    shared::{ControllerSignal, ElectricalBusType, ElectricalBuses, PneumaticValve},
+    simulation::{SimulationElement, SimulationElementVisitor, UpdateContext},
+};
+
+use uom::si::{
+    f64::*,
+    pressure::{pascal, psi},
+    ratio::{percent, ratio},
+    temperature_interval,
+    thermodynamic_temperature::kelvin,
+    volume_rate::cubic_meter_per_second,
+};
+
+use super::{ControllablePneumaticValve, PneumaticContainer, PneumaticValveSignal};
+
+/// A valve only controlled by the physical forces due to the pressure gradient. This does not accept any signals.
+pub struct PurelyPneumaticValve {
+    open_amount: Ratio,
+    connector: PneumaticContainerConnector,
+    spring_characteristic: f64,
+}
+impl PurelyPneumaticValve {
+    pub fn new(spring_characteristic: f64) -> Self {
+        Self {
+            open_amount: Ratio::new::<ratio>(0.),
+            connector: PneumaticContainerConnector::new(),
+            spring_characteristic,
+        }
+    }
+
+    pub fn update_move_fluid(
+        &mut self,
+        context: &UpdateContext,
+        container_one: &mut impl PneumaticContainer,
+        container_two: &mut impl PneumaticContainer,
+    ) {
+        self.set_open_amount_from_pressure_difference(
+            container_one.pressure() - container_two.pressure(),
+        );
+
+        self.connector
+            .with_transfer_speed_factor(self.open_amount)
+            .update_move_fluid(context, container_one, container_two);
+    }
+
+    fn set_open_amount_from_pressure_difference(&mut self, pressure_difference: Pressure) {
+        self.open_amount = Ratio::new::<ratio>(
+            2. / PI
+                * (pressure_difference.get::<psi>() * self.spring_characteristic)
+                    .atan()
+                    .max(0.),
+        );
+    }
+
+    pub fn fluid_flow(&self) -> VolumeRate {
+        self.connector.fluid_flow()
+    }
+
+    pub fn open_amount(&self) -> Ratio {
+        self.open_amount
+    }
+}
+impl PneumaticValve for PurelyPneumaticValve {
+    fn is_open(&self) -> bool {
+        self.open_amount.get::<percent>() > 0.
+    }
+}
+
+pub struct ElectroPneumaticValve {
```

Comment:
Yeah, I've thought about this, but the solutions I've considered seem a bit like forced abstractions where modifications in the future would be harder because they're tied together too strictly.